plugin.video.ampya/default.py

```diff
@@ -0,0 +1,341 @@
+  liz.addContextMenuItems( commands )  
+  ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz)
+  return ok
+
+
+def getUrl(url,data="x"):        
+        debug("Get Url: " +url)
+        opener = urllib2.build_opener(urllib2.HTTPCookieProcessor())
+        userAgent = "Dalvik/2.1.0 (Linux; U; Android 5.0;)"
+        opener.addheaders = [('User-Agent', userAgent)]
+        try:
+          if data!="x" :
+             content=opener.open(url,data=data).read()
+          else:
+             content=opener.open(url).read()
+        except urllib2.HTTPError as e:
+             #print e.code   
+             cc=e.read()  
+             struktur = json.loads(cc)  
+             error=struktur["errors"][0] 
+             error=unicode(error).encode("utf-8")
+             debug("ERROR : " + error)
+             dialog = xbmcgui.Dialog()
+             nr=dialog.ok("Error", error)
+             return ""
+             
+        opener.close()
+        return content
+
+
+      #addDir(namenliste[i], namenliste[i], mode+datum,logoliste[i],ids=str(idliste[i]))
+   #xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True)   
+
+def ListeChannels(url):
+  content=getUrl(url)
+  struktur = json.loads(content)
+  for element in struktur:   
+    title=element["asset"]["title"]
+    kuenstler_name=element["asset"]["display_artist_title"]
+    kuenstler_id=element["asset"]["artist_id"]
+    video_file_id=element["asset"]["video_file_id"]
+    Token=element["asset"]["token"]
+    duration=element["asset"]["duration"]
+    iconimage=getbild(video_file_id)
+    id=element["asset"]["id"]
+
+    
+    
+    debug("VIDEOID: "+str(video_file_id))
+    debug("iconimage :"+iconimage)
+    
+    
+    addLink(title +"("+kuenstler_name+")", Token, "playvideo", iconimage, duration=duration, desc="",artist_id=kuenstler_id,liedid=id)
+    debug("--------------")
+  xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True)   
+
+def getbild(video_file_id):
+    longid="%07d"%video_file_id
+    einsbisdrei=str(longid)[0:5]
+    iconimage="http://files.putpat.tv/artwork/posterframes/"+ einsbisdrei +"/" + longid +"/v"+longid+"_posterframe_putpat_large.jpg"
+    return iconimage
+    
+def ListeVideos(url):
+  content=getUrl(url)
+  struktur = json.loads(content)
+  for element in struktur:   
+    title=element["asset"]["title"]
+    kuenstler_name=element["asset"]["display_artist_title"]
+    kuenstler_id=element["asset"]["artist_id"]
+    video_file_id=element["asset"]["video_file_id"]
+    Token=element["asset"]["token"]
+    #duration=element["asset"]["duration"]
+    iconimage=getbild(video_file_id)
+    id=element["asset"]["id"]
+    
+    debug("VIDEOID: "+str(video_file_id))
+    debug("iconimage :"+iconimage)
+    
+    
+    addLink(title +"("+kuenstler_name+")", Token, "playvideo", iconimage, desc="",artist_id=kuenstler_id,liedid=id)
```

Comment:
This is not a blocker but, since you're contributing a lot lately, you can get huge performance improvements (especially if you're adding a big list of videos) if you create a list of listitems and call  xbmcplugin.addDirectoryItems directly instead of calling xbmcplugin.addDirectoryItem for each listitem. (https://codedocs.xyz/xbmc/xbmc/group__python__xbmcplugin.html#ga3a00602b6bcfdc77ca0c56362c84540c)